logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+		rc = sqlite3_step(stmt_check_if_BLOBS_TABLE_exists);
+		if (unlikely(rc != SQLITE_ROW)) fatal_sqlite3_err(rc, __LINE__); /* COUNT(*) query should always return SQLITE_ROW */
+		
+		/* If BLOBS_TABLE doesn't exist, create and populate it */
+		if(sqlite3_column_int(stmt_check_if_BLOBS_TABLE_exists, 0) == 0){
+			
+			/* 1. Create it */
+			rc = sqlite3_exec(p_file_infos_arr->data[i]->db,
+                      "CREATE TABLE IF NOT EXISTS " BLOBS_TABLE "("
+                      "Id 		INTEGER 	PRIMARY KEY,"
+                      "Filename	TEXT		NOT NULL,"
+                      "Filesize INTEGER 	NOT NULL"
+                      ");",
+                      0, 0, &err_msg);
+			if (unlikely(rc != SQLITE_OK)) {
+				sqlite3_free(err_msg);
+				fatal("Failed to create %s. SQL error: %s", BLOBS_TABLE, err_msg);
+			} 
+			// else debug(D_LOGS_MANAG, "Table %s created successfully\n", BLOBS_TABLE);
+			
+			/* 2. Populate it */
+			sqlite3_stmt *stmt_init_BLOBS_table;
+			rc = sqlite3_prepare_v2(p_file_infos_arr->data[i]->db,
+                            "INSERT INTO " BLOBS_TABLE 
+                            " (Filename, Filesize) VALUES (?,?) ;",
+                            -1, &stmt_init_BLOBS_table, NULL);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+			for( int i = 0; i < BLOB_MAX_FILES; i++){
+				char *filename = mallocz(snprintf(NULL, 0, BLOB_STORE_FILENAME ".%d", i) + 1);
+				sprintf(filename, BLOB_STORE_FILENAME ".%d", i);
+				rc = sqlite3_bind_text(stmt_init_BLOBS_table, 1, filename, -1, NULL);
+                if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);
+                rc = sqlite3_bind_int64(stmt_init_BLOBS_table, 2, (sqlite3_int64) 0);
+                if (rc != SQLITE_OK) fatal_sqlite3_err(rc, __LINE__);      
+				rc = sqlite3_step(stmt_init_BLOBS_table);
+				if (rc != SQLITE_DONE) fatal_sqlite3_err(rc, __LINE__);
+				rc = sqlite3_reset(stmt_init_BLOBS_table);
+				if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+				freez(filename);
+			}
+			rc = sqlite3_finalize(stmt_init_BLOBS_table);
+			if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+		}
+		rc = sqlite3_finalize(stmt_check_if_BLOBS_TABLE_exists);
+		if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+		
+		/* If LOGS_TABLE doesn't exist, create it */
+		rc = sqlite3_exec(p_file_infos_arr->data[i]->db,
+                      "CREATE TABLE IF NOT EXISTS " LOGS_TABLE "("
+                      "Id 					INTEGER 	PRIMARY KEY,"
+                      "FK_BLOB_Id			INTEGER		NOT NULL,"
+                      "BLOB_Offset			INTEGER		NOT NULL,"
+                      "Timestamp 			INTEGER		NOT NULL,"
+                      "Msg_compr_size 		INTEGER		NOT NULL,"
+                      "Msg_decompr_size 	INTEGER		NOT NULL,"
+                      "FOREIGN KEY (FK_BLOB_Id) REFERENCES " BLOBS_TABLE " (Id) ON DELETE CASCADE ON UPDATE CASCADE"
```

Comment:
Ping @stelfrag , we have now foreign key and the expected ACID! :rofl:  :heart:

Reply:
:)